frontend/src/metabase/dashboard/hoc/controls/use-dashboard-url-params.ts

```diff
@@ -0,0 +1,140 @@
+import type { Location } from "history";
+import { useCallback, useEffect, useMemo } from "react";
+import { replace } from "react-router-redux";
+import { useMount, usePrevious } from "react-use";
+import { isEqual } from "underscore";
+
+import type { DashboardUrlHashOptions } from "metabase/dashboard/hoc/controls/types";
+import type { DashboardDisplayOptionControls } from "metabase/dashboard/hoc/controls/use-dashboard-display-options";
+import { useDashboardDisplayOptions } from "metabase/dashboard/hoc/controls/use-dashboard-display-options";
+import { parseHashOptions, stringifyHashOptions } from "metabase/lib/browser";
+import { useDispatch } from "metabase/lib/redux";
+import { isNotFalsy } from "metabase/lib/types";
+
+const removeEmptyOptions = (obj: Record<string, unknown>) => {
+  return Object.fromEntries(
+    Object.entries(obj).filter(([_, v]) => isNotFalsy(v)),
+  );
+};
+
+export const useDashboardUrlParams = ({
+  location,
+  onRefresh,
+}: {
+  location: Location;
+  onRefresh: () => Promise<void>;
+}): DashboardDisplayOptionControls & {
+  loadDashboardParams: () => void;
+} => {
+  const dispatch = useDispatch();
+
+  const {
```

Comment:
In the static dashboard in the SDK, we can use this hook like this without the `location`